indexer/processor/processor.go

```diff
@@ -0,0 +1,68 @@
+package processor
+
+import (
+	"time"
+
+	"github.com/ethereum-optimism/optimism/indexer/database"
+	"github.com/ethereum-optimism/optimism/indexer/node"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const defaultLoopInterval = 5 * time.Second
+
+// processFn is the the function used to process unindexed headers. In
+// the event of a failure, all database operations are not committed
+type processFn func(*database.DB, []*types.Header) error
+
+type processor struct {
+	fetcher *node.Fetcher
+
+	db         *database.DB
+	processFn  processFn
+	processLog log.Logger
+}
+
+// Start kicks off the processing loop. This is a blocking operation and should be run within its own goroutine
+func (p processor) Start() {
```

Comment:
What does op-node use for this?

Reply:
Good question, not entirely sure cc @tynes

Reply:
Looked into op-node. op-node/ defines a driver for walking the L1 chain. It's opinionated and specifically build for the op-node use case. This is simple enough for us to define here